fix(flags): Improve UX around disabled actions/fields

dustinbyrne opened the pull request:
## Problem

When trying to edit a feature flag but the user is lacking permissions or a linked experiment prevents certain changes, user feedback has revealed that it's not clear _why_ an edit cannot be made.

## Changes

1. Users who cannot edit a flag now see the `edit` button disabled with a reason. The message makes it clear that they do not have the proper permissions to edit the flag.

2. When a flag is linked to an experiment, editing variants is disabled. The variant text input now has a hover message stating the reason why it cannot be edited.

Trying to edit a flag as a read-only user
<img width="431" height="110" alt="image" src="https://github.com/user-attachments/assets/9679056b-9a88-4cd6-a3db-f57d9dc11f85" />

Trying to change the control variant key when linked to a draft experiment
<img width="787" height="189" alt="image" src="https://github.com/user-attachments/assets/3d7b8bdb-23f3-4f6e-9c68-f0c1035ea3f3" />

Trying to edit variant keys when linked to a launched experiment
<img width="792" height="241" alt="image" src="https://github.com/user-attachments/assets/0efff586-4b04-4291-8b49-db9eeb92aaf8" />

## How did you test this code?

Manual testing, see images above

dustinbyrne committed:
fix(flags): Improve UX around disabled actions/fields

1. Users who cannot edit a flag now cannot click the `edit` button. This
makes it clear that they do not have the proper permissions

vs.

2. When a flag is linked to an experiment, editing variants is disabled.
The variant text input now has a hover message stating the reason why it
cannot be edited.

gustavohstrassburger reviewed (approved):
Haven't tested it, but I trust you.

dustinbyrne committed:
style: Add a period to close out disabled reasons

dustinbyrne committed:
feat(lemon-ui): Allow disabledReason to be marked as interactive

This adds an interactive grace period for the tooltip where it doesn't
immediately close if the mouse exists the hoverable area briefly
(750ms). This prevents a frustrating experience where you're trying to
mouse over the tooltip but it disappears.

dustinbyrne committed:
fix: Link to the experiment in the tooltip

dustinbyrne committed:
style: Specify `variant keys`
